thrust/thrust/system/omp/detail/scan.h

```diff
@@ -106,7 +106,9 @@ OutputIterator scan_impl(
 
     if (start < n)
     {
-      block_sums[tid] = ::cuda::std::reduce(first + start, first + end, accum_t{}, wrapped_binary_op);
+      // For both has_init and no-init cases: reduce each block using first element as init
+      accum_t first_elem = *(first + start);
+      block_sums[tid]    = ::cuda::std::reduce(first + start + 1, first + end, first_elem, wrapped_binary_op);
```

Comment:
I am also wondering what happens if `num_threads` becomes larger than `parallel_scan_threshold`. In this case `block_size` becomes zero and `start` and `end` are the same. I think we should catch this sooner where we have:
```c++
if (static_cast<size_t>(n) < parallel_scan_threshold || num_threads <= 1)
```
and do:
```c++
if (static_cast<size_t>(n) < ::cuda::std::max(parallel_scan_threshold, num_threads) || num_threads <= 1)
```